external_parser/parse_example_converter.cc

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) by respective owners including Yahoo!, Microsoft, and
+// individual contributors. All rights reserved. Released under a BSD (revised)
+// license as described in the file LICENSE.
+
+#include "parse_example_converter.h"
+#include "example.h"
+#include "global_data.h"
+
+namespace VW {
+namespace external {
+
+binary_json_converter::binary_json_converter(std::unique_ptr<i_joiner> &&joiner)
+    : _parser(std::move(joiner)) {}
+
+binary_json_converter::~binary_json_converter() = default;
+
+bool binary_json_converter::parse_examples(vw *all,
+                                           v_array<example *> &examples) {
+  while (_parser.parse_examples(all, examples)) {
```

Comment:
Won't this array grown unbounded?

Reply:
no it gets cleaned up, see [here](https://github.com/VowpalWabbit/reinforcement_learning/pull/358/files#diff-36a22962013a321a397c4ef1a1cb47d690b407e35e9bd49d144913dc64bad082R463)